### PR TITLE
Unblock local development with Java Cards

### PR DESCRIPTION
**Pull request by arsalansufi**
_Reviewing by commit recommended_

## Overview

Issue link: https://github.com/votingworks/vxsuite/issues/3104

This PR unblocks local development with Java Cards! It:

- Adds a script for programming a first dev sys admin Java Card to bootstrap all other card programming
- Adds an `ENABLE_JAVA_CARDS` feature flag
- Configures backends for Java Card auth when that flag is enabled

I'll share a quick start guide with everyone in Slack but have also documented everything (minus the feature flag) in the `libs/auth` README.

## Script Demo

Helpful error message:

```
$ ./scripts/program-dev-system-administrator-java-card 
❌ Received response APDU with non-success status word: 00 a4

Have you run `./scripts/configure-dev-java-card` on this card yet?
If not, that's likely the cause of this error.
Run that and then retry.
```

You've already seen this script (not the new wrapper but the base script), but I'm including it here for completeness:

```
$ ./scripts/configure-dev-java-card
----------------------------------------------
🧹 Uninstalling applet if already installed...
----------------------------------------------
-----------------------
💿 Installing applet...
-----------------------
-------------------------------------------
🔧 Running applet configuration commands...
-------------------------------------------
Sending the following APDUs over a GlobalPlatform Secure Channel:
> 0c:24:ff:80:08:30:30:30:30:30:30:ff:ff
> 0c:24:ff:81:08:00:00:00:00:00:00:00:00
> 0c:db:3f:00:07:68:05:a0:03:86:01:0f
> 0c:db:3f:00:14:66:12:8b:01:f0:8c:01:7f:8d:01:00:8e:01:11:8f:01:04:90:01:00
> 0c:db:3f:00:14:66:12:8b:01:f1:8c:01:12:8d:01:00:8e:01:11:8f:01:04:90:01:00
> 0c:db:3f:00:0d:64:0b:8b:03:5f:c1:f0:8c:01:7f:8d:01:00
> 0c:db:3f:00:0d:64:0b:8b:03:5f:c1:f1:8c:01:7f:8d:01:00
> 0c:db:3f:00:0d:64:0b:8b:03:5f:c1:f2:8c:01:7f:8d:01:00
> 0c:db:3f:00:0d:64:0b:8b:03:5f:c1:f3:8c:01:7f:8d:01:00
> 0c:db:3f:00:0d:64:0b:8b:03:5f:c1:f4:8c:01:7f:8d:01:00
> 0c:db:3f:00:0d:64:0b:8b:03:5f:c1:f5:8c:01:7f:8d:01:00
------------------------------------------------
🔏 Creating and storing card VotingWorks cert...
------------------------------------------------
--------
✅ Done!
--------
```

Take 2:

```
$ ./scripts/program-dev-system-administrator-java-card 
✅ Done!
```

Party time! :tada:

## Testing

- [x] Tested the dev sys admin Java Card programming script (see above)
- [x] Enabled the feature flag and verified that I could use a Java Card for machine auth (after running the above scripts)

## Checklist

- [ ] ~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced~ N/A
- [x] I have added JSDoc comments to any newly introduced exports

### CODE REVIEW

**Review by benadida — APPROVED** [review]
reviewed mostly for security, but not seeing anything in other aspects that's at all controversial :)

File: apps/scan/backend/src/server.ts [hunk 1]
```diff
@@ -39,7 +52,12 @@ export function start({
     },
     logger,
   });
+
+  // Clear any cached data
+  workspace.clearUploads();
```
[the remark this answers is not in the document]
Comment (arsalansufi) [reply apps/scan/backend/src/server.ts:57]: Ah good q - it's not! Just a line that was moved to match the order of operations in other backends

moved

[the remark this answers is not in the document]
Comment (benadida) [reply apps/scan/backend/src/server.ts:57]: oh no you're being generous, it's not a good question, I should have looked more closely :)

File: libs/auth/src/java_card.ts [hunk 2]
```diff
@@ -314,13 +313,7 @@ export class JavaCard implements Card {
...
         MAX_DATA_OBJECT_SIZE_BYTES * i + MAX_DATA_OBJECT_SIZE_BYTES
       );
-      await this.putData(
-        objectId,
-        Buffer.concat([
-          constructTlv(PUT_DATA.TAG_LIST_TAG, objectId),
-          constructTlv(PUT_DATA.DATA_TAG, chunk),
-        ])
-      );
+      await this.putData(objectId, chunk);
```
Comment (arsalansufi) [inline libs/auth/src/java_card.ts:316]: I'd forgotten that `putData` internally handles wrapping the data

File: libs/auth/src/java_card.ts [hunk 3]
```diff
@@ -576,15 +570,17 @@ export class JavaCard implements Card {
...
+    const dataTlv = await this.cardReader.transmit(
       new CardCommand({
         ins: GET_DATA.INS,
         p1: GET_DATA.P1,
         p2: GET_DATA.P2,
         data: constructTlv(GET_DATA.TAG_LIST_TAG, objectId),
       })
     );
+    const [, , data] = parseTlv(PUT_DATA.DATA_TAG, dataTlv);
+    return data;
```
Comment (arsalansufi) [inline libs/auth/src/java_card.ts:583]: This makes it such that there's parity between `putData` and `getData`. The data provided to `putData` is now exactly what you get back from `getData`.

**Comment by arsalansufi** [comment]
@jonahkagan I'm gonna go ahead and merge this but please feel free to leave comments if there's anything you'd like for me to adjust when you get back Monday or also just ignore - up to you!

File: libs/auth/src/java_card.ts (hunk 2 above)
Comment (jonahkagan) [reply libs/auth/src/java_card.ts:316]: Do we need a regression test for this bug?

Comment (arsalansufi) [reply libs/auth/src/java_card.ts:316]: Yes we do! I'm incorporating this into the tests I'm still working on for the JavaCard class

**Pull request merged**